### PR TITLE
chore: documentation on orchestrator sub module and the API.

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-6074
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Review by kjaskiewiczz — COMMENTED** [review]
I'm a bit lost - app update module looks like some abstraction layer to different modules (called sub-modules), but the sub-module API is specific for containers, so it doesn't seem it will be very universal.

**Comment by merlin-northern** [comment]
> I'm a bit lost - app update module looks like some abstraction layer to different modules (called sub-modules), but the sub-module API is specific for containers, so it doesn't seem it will be very universal.

let's discuss in a call. (you use the same API do deploy a standalone app based on tar.gz, with some strange nomenclature ;-))

File: docs/README-submodule-api.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+The Artifact as created by the generator, carries a certain amount of metadata, used to properly
+handle all the above calls.
+
+```json
+{
+  "version": "1.0",
+  "platform": "linux/arm/v7",
+  "application_name": "myapp0",
+  "flavour": "installable-application",
```
Comment (kjaskiewiczz) [inline docs/README-submodule-api.md:244]: I think the version is enough; if we'll decide to change/extend implementation we can simply bump the version

**Review by kjaskiewiczz — COMMENTED** [review]
The description of high level operations play nice with idea of generic application update module, but the sub-module API still looks quite container specific.
I'd prefer the whole update module to be container specific, because it would be more clear to me.
What exactly do we gain by providing generic update module for applications instead of having containers-update-modules with docker-compose and k8s submodules?

File: docs/README-submodule-api.md (hunk 1 above)
Comment (merlin-northern) [reply docs/README-submodule-api.md:244]: sure.

**Comment by merlin-northern** [comment]
> [...] What exactly do we gain by providing generic update module for applications instead of having containers-update-modules with docker-compose and k8s submodules?

I think we should think in terms of applications, as a basic unit of abstraction. I think that how we should look at this: application updates with support for containers in the first iteration. you do not think it makes sense?

**Comment by kjaskiewiczz** [comment]
> I think that how we should look at this: application updates with support for containers in the first iteration. you do not think it makes sense?

I understand and agree with the idea. What doesn't really work for me is the API which looks container specific, so, IMO, we should have generic module with generic API or specific module with specific API. To me it looks like with this PR we're introducing generic module with specific (containers specific) API.
I'm not sure how the generic API should look like, and this is why my personal preference would be to simply change the module to containers-update-module and keep specific API, even from before your last change, but
I'm not planning to block the work here (in other words I'm giving up and approving ;) ).

**Comment by merlin-northern** [comment]
> > I think that how we should look at this: application updates with support for containers in the first iteration. you do not think it makes sense?
> 
> I understand and agree with the idea. What doesn't really work for me is the API which looks container specific, so, IMO, we should have generic module with generic API or specific module with specific API. To me it looks like with this PR we're introducing generic module with specific (containers specific) API. I'm not sure how the generic API should look like, and this is why my personal preference would be to simply change the module to containers-update-module and keep specific API, even from before your last change, but I'm not planning to block the work here (in other words I'm giving up and approving ;) ).

since we are discussing a lot behind the scenes I updated the API, so now it looks more general. I hate when you say "I give up".

**Pull request merged**